### ISSUE TITLE
Unmarshalled JSON with types.MatchQuery returns double-quoted string fields

### ISSUE THREAD

**Issue by Hexta** [elastic/go-elasticsearch#684]
Unmarshalling JSON with types.MatchQuery fields returns double quoted strings for fields with `string` type.

```go
mq := types.MatchQuery{Query: "value"}
mqb, _ := json.MarshalIndent(mq, "", "  ")

fmt.Println(string(mqb))
fmt.Printf("%#v\n", mq.Query)

_ = json.Unmarshal(mqb, &mq)
fmt.Printf("%#v\n", mq.Query)
```

Output:
```plain
{
  "query": "value"
}
"value"
"\"value\""
```

I think, the issue is in [custom unmarshal code](https://github.com/elastic/go-elasticsearch/blob/4ea9db9cf63079fd1dbdc92e44737869f40589d9/typedapi/types/matchquery.go#L212)

```go
case "query":
	var tmp json.RawMessage
	if err := dec.Decode(&tmp); err != nil {
		return err
	}
	o := string(tmp)
	s.Query = o
```

`tmp` variable should have `string` type, not `json.RawMessage`.

**Comment by Anaethelion** [comment]
Hi @Hexta 

Thank you for reporting this.
The situation is a bit more complex due to some special case for deserialization.
I'm already working on a fix, will report back when its updated!

**Comment by Anaethelion** [comment]
Fixed in release [release/8.8.2](https://github.com/elastic/go-elasticsearch/releases/tag/v8.8.2)
